Mathlib/Tactic/ExtractGoal.lean

```diff
@@ -116,7 +117,11 @@ elab (name := extractGoal) "extract_goal" full?:&"*"? name?:(colGt ppSpace ident
              else mkAuxName ((← getCurrNamespace) ++ `extracted) 1
   let msg ← withoutModifyingEnv <| withoutModifyingState do
     let mut g ← getMainGoal
-    unless full?.isSome do
+    let mut cleanup := full?.isNone
+    if (← g.getType >>= instantiateMVars).consumeMData.isConstOf ``False then
```

Comment:
Would there be some difference to using
```lean
if ← Meta.isDefEq (← g.getType') (.const ``False []) then
```
instead?

Reply:
I wanted to see `False` in the goal directly, not after unfolding, since that's a strong indicator that this is a contradiction proof. One reason not to unfold is that the goal might include variables that disappear from it after unfolding.